### PR TITLE
Add Red Brick Road Offline Handling to Payments List

### PR DESCRIPTION
**Pull request by thienlnam**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/221881
### Tests

1. Modify your Onyx data to include an error in the bank accounts
2. First start by exposing Onyx to the window by adding `window.Onyx = Onyx;` to setup/index.js
3. Then in the console, you should be able to do something like below with an existing bank account
`Onyx.merge('bankAccountList', {"827591": {pendingAction: 'delete', errors: {1: 'This is an error'}}})`
4. Verify that when you go to Settings, you see the error dot on the 'Payments' tab like below, and when you click on the payments tab you see the error
5. Verify you can click on the x to remove the error

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

- [ ] Verify that no errors appear in the JS console

#### Web

Screen Shot 2022-08-09 at 6 05 53 PM

Screen Shot 2022-08-09 at 6 13 28 PM

Screen Shot 2022-08-09 at 6 47 05 PM

### CODE REVIEW

**Comment by aldo-expensify** [comment]
hmm whats with Melvin repeating this comment: https://github.com/Expensify/App/pull/10333#issuecomment-1210059888

File: package-lock.json [hunk 1]
```diff
@@ -37719,9 +37719,9 @@
           "integrity": "sha512-72fSenhMw2HZMTVHeCA9KCmpEIbzWiQsjN+BHcBbS9vr1mtt+vJjPdksIBNUmKAW8TFUDPJK5SUU3QhE9NEXDw=="
         },
         "electron-to-chromium": {
-          "version": "1.4.211",
-          "resolved": "https://registry.npmjs.org/electron-to-chromium/-/electron-to-chromium-1.4.211.tgz",
-          "integrity": "sha512-BZSbMpyFQU0KBJ1JG26XGeFI3i4op+qOYGxftmZXFZoHkhLgsSv4DHDJfl8ogII3hIuzGt51PaZ195OVu0yJ9A=="
+          "version": "1.4.213",
+          "resolved": "https://registry.npmjs.org/electron-to-chromium/-/electron-to-chromium-1.4.213.tgz",
+          "integrity": "sha512-+3DbGHGOCHTVB/Ms63bGqbyC1b8y7Fk86+7ltssB8NQrZtSCvZG6eooSl9U2Q0yw++fL2DpHKOdTU0NVEkFObg=="
```
Comment (aldo-expensify) [inline package-lock.json:37724]: Are you intentionally updating this package version?

File: src/libs/actions/PaymentMethods.js [hunk 2]
```diff
@@ -311,6 +312,42 @@ function dismissSuccessfulTransferBalancePage() {
...
 
+/**
+ * Looks through each payment method to see if there is an existing error
+ * @param {Object} bankList
+ * @param {Object} cardList
+ * @returns {Boolean}
+ */
+function hasPaymentMethodError(bankList, cardList) {
+    const combinedPaymentMethods = lodashMerge(bankList, cardList);
+    return _.some(combinedPaymentMethods, item => !_.isEmpty(item.errors));
```
Comment (aldo-expensify) [inline src/libs/actions/PaymentMethods.js:323]: Why do we check the cardList if we didn't add the `errors` and `pendingAction` keys [here](https://github.com/Expensify/App/blob/2debc84873eb3b463c9b34e2ec81ad2f83e50d66/src/libs/PaymentUtils.js#L66-L82) like you did with the bank account items? Is that intentional?

From the looks of the code, it looks like it was written generic enough to handle bank accounts and cards... so maybe you missed adding those keys in PaymentUtils/cards?

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 3]
```diff
@@ -164,6 +167,20 @@ class PaymentMethodList extends Component {
         return combinedPaymentMethods;
     }
 
+    /**
+     * Dismisses the error on the payment method
+     * @param {*} item
+     */
+    dismissError(item) {
+        const paymentList = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST;
+        const paymentID = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? lodashGet(item, ['accountData', 'bankAccountID']) : lodashGet(item, ['accountData', 'cardID']);
```
Comment (aldo-expensify) [inline src/pages/settings/Payments/PaymentMethodList.js:176]: I see you are using `lodashGet` to get `['accountData', 'bankAccountID']`, so that makes me think that you are considering these things could be missing... but if they are missing you don't have a default, so you get `undefined`, right? what happens then if you call `PaymentMethods.clearDeletePaymentMethodError` or `PaymentMethods.clearAddPaymentMethodError` with an `undefined` paymentID? I don't think it crashes, but you would a weird key `undefined` in Onyx.

My advice would be:
- remove `lodashGet` and let it crash if the info was not there, or
- add a check `if(paymentID){return;}` to exit if it didn't find it (and log?)

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
Looking good so far! the trick to insert the error in Onyx is very clever and made testing easier.. thanks!

I added the error as suggested and disabled the network. I think the item text should have been strikethrough, but it didn't work:

image

Maybe you need to do something like @iwiznia did [here](https://github.com/Expensify/App/pull/9931/files#diff-2584464930cd870e953ab50a7dbe0dac4b9d9f6c2818f61fb1dc06e3828dd897) (add `style` prop to `MenuItem` in your case... but that may be a bit confusing since there are already other styles there 🤮 )

**Comment by iwiznia** [comment]
> hmm whats with Melvin repeating this comment: #10333 (comment)

Probably we are not checking if we already left the comment, but maybe we should not fix that, because Melvin commented 42 times, but the PR still contains changes in the lock, when it shouldn't 😄

File: src/libs/PaymentUtils.js [hunk 4]
```diff
@@ -58,6 +58,8 @@ function formatPaymentMethods(bankAccountList, cardList, payPalMeUsername = '',
             accountType: CONST.PAYMENT_METHODS.BANK_ACCOUNT,
             accountData: _.extend({}, bankAccount, {icon}),
             isDefault,
+            errors: bankAccount.errors || null,
+            pendingAction: bankAccount.pendingAction || null,
```
Comment (iwiznia) [inline src/libs/PaymentUtils.js:62]: I don't think you need the `|| null`

File: src/libs/actions/PaymentMethods.js [hunk 5]
```diff
@@ -311,6 +312,42 @@ function dismissSuccessfulTransferBalancePage() {
...
+}
+
+/**
+ * Clears the error for the specified payment item
+ * @param {String} paymentList The onyx key for the provided payment method
```
Comment (iwiznia) [inline src/libs/actions/PaymentMethods.js:328]: This is a weird param name, if it is named xList, is reasonable to assume it will be an array, but here is really a string for the key. IMO we should change this (same below)

File: src/pages/settings/InitialSettingsPage.js [hunk 6]
```diff
@@ -78,39 +87,6 @@ const defaultProps = {
     ...withCurrentUserPersonalDetailsDefaultProps,
 };
 
-const defaultMenuItems = [
```
Comment (iwiznia) [inline src/pages/settings/InitialSettingsPage.js:81]: Why was this moved?

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 7]
```diff
@@ -164,6 +167,20 @@ class PaymentMethodList extends Component {
         return combinedPaymentMethods;
     }
 
+    /**
+     * Dismisses the error on the payment method
+     * @param {*} item
```
Comment (iwiznia) [inline src/pages/settings/Payments/PaymentMethodList.js:172]:
```suggestion
     * @param {Object} item
```

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 8]
```diff
@@ -164,6 +167,20 @@ class PaymentMethodList extends Component {
...
+     */
+    dismissError(item) {
+        const paymentList = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST;
+        const paymentID = item.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? lodashGet(item, ['accountData', 'bankAccountID']) : lodashGet(item, ['accountData', 'cardID']);
+        if (item.pendingAction === 'delete') {
```
Comment (iwiznia) [inline src/pages/settings/Payments/PaymentMethodList.js:177]: There's a constant for this [here](https://github.com/Expensify/App/blob/15f7fa622805ee2971808d6bc67181c4715f0c62/src/CONST.js#L777)

**Review by iwiznia — CHANGES_REQUESTED** [review]

**Comment by iwiznia** [comment]
I think this is missing adding the dot to the avatar, no?

File: src/pages/settings/InitialSettingsPage.js (hunk 6 above)
Comment (aldo-expensify) [reply src/pages/settings/InitialSettingsPage.js:81]: The `brickRoadIndicator: PaymentMethods.hasPaymentMethodError(props.bankAccountList, props.cardList) ? 'error' : null,` needs access to something in `props` now

File: src/libs/actions/PaymentMethods.js (hunk 2 above)
Comment (thienlnam) [reply src/libs/actions/PaymentMethods.js:323]: Good catch, I just missed adding those keys for the cards

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 3 above)
Comment (thienlnam) [reply src/pages/settings/Payments/PaymentMethodList.js]: I decided to just add a check if paymentID doesn't exist - We shouldn't let crashes happen but i imagine if the user presses this quickly twice the paymentID would be empty

File: src/libs/actions/PaymentMethods.js (hunk 5 above)
Comment (thienlnam) [reply src/libs/actions/PaymentMethods.js]: Good point, I originally was passing the payment list but then realized I only needed to pass the onyx key - updated

**Comment by thienlnam** [comment]
> I think this is missing adding the dot to the avatar, no?

Yup, added that in the latest changes

> Are you intentionally updating this package version?
> Probably we are not checking if we already left the comment, but maybe we should not fix that, because Melvin commented 42 times, but the PR still contains changes in the lock, when it shouldn't 😄

I already did an NPM I which is how I ended up with those changes so melvin didn't help me here 😞 

> Maybe you need to do something like @iwiznia did [here](https://github.com/Expensify/App/pull/9931/files#diff-2584464930cd870e953ab50a7dbe0dac4b9d9f6c2818f61fb1dc06e3828dd897) (add style prop to MenuItem in your case... but that may be a bit confusing since there are already other styles there 🤮 )

I wonder if we should automatically merge styles in the offlinewithfeedback component instead of on the menu item and option row. I did end up adding it for now but it would probably be more efficient since we're already recursively adding it onto children we could just merge the existing styles

**Comment by iwiznia** [comment]
>I already did an NPM I which is how I ended up with those changes so melvin didn't help me here 😞

I don't understand what you meant here, but please undo the changes in the lock file

File: src/components/OfflineWithFeedback.js [hunk 9]
```diff
@@ -44,19 +44,17 @@ const propTypes = {
         PropTypes.object,
     ]),
 
-    /** Additional styles to add after local styles.  Applied to the error text portion */
-    errorStyle: PropTypes.oneOfType([
-        PropTypes.arrayOf(PropTypes.object),
-        PropTypes.object,
-    ]),
+    /** Additional style object for the error row */
+    errorRowStyles: PropTypes.arrayOf(PropTypes.object),
```
Comment (iwiznia) [inline src/components/OfflineWithFeedback.js:48]: Why did you make this stricter? Styles normally can be passed as an array of objects or as one object, why force to have an array always in this case?

File: src/libs/PaymentUtils.js [hunk 10]
```diff
@@ -79,6 +79,8 @@ function formatPaymentMethods(bankAccountList, cardList, payPalMeUsername = '',
             accountType: CONST.PAYMENT_METHODS.DEBIT_CARD,
             accountData: _.extend({}, card, {icon}),
             isDefault,
+            errors: card.errors,
```
Comment (iwiznia) [inline src/libs/PaymentUtils.js:82]: NAB for this PR, but isn't all this code data massaging and thus something we should not be doing?

**Comment by iwiznia** [comment]
Oh you also got conflicts

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 11]
```diff
@@ -185,20 +209,22 @@ class PaymentMethodList extends Component {
...
-                    disabled={item.disabled}
-                    iconFill={item.iconFill}
-                    iconHeight={item.iconSize}
-                    iconWidth={item.iconSize}
-                    badgeText={this.getDefaultBadgeText(item.isDefault)}
-                    wrapperStyle={item.wrapperStyle}
-                    shouldShowSelectedState={this.props.shouldShowSelectedState}
-                    isSelected={this.props.selectedMethodID === item.methodID}
-                />
+                <OfflineWithFeedback onClose={() => this.dismissError(item)} pendingAction={item.pendingAction} errors={item.errors} errorRowStyles={styles.ph6}>
```
Comment (MariaHCD) [inline src/pages/settings/Payments/PaymentMethodList.js:212]:
```suggestion
                <OfflineWithFeedback onClose={() => this.dismissError(item)} pendingAction={item.pendingAction} errors={item.errors} errorRowStyles={[styles.ph6]}>
```

I think this needs to be an array since its defined [here](https://github.com/Expensify/App/blob/2c391e4b96de35f19241fd93252e0fdccb7687bd/src/components/OfflineWithFeedback.js#L47-L48) as an array. I get this error when testing:

Screen Shot 2022-08-11 at 4 11 23 PM

**Review by MariaHCD — COMMENTED** [review]
I don't think its being caused by this PR but just noting that a P2P debit card seems to be stored in `bankAccountList` as well as the `cardList`:

Screen Shot 2022-08-11 at 4 22 44 PM

The data from the API looks to be correct, though:

Screen Shot 2022-08-11 at 4 27 21 PM

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 11 above)
Comment (iwiznia) [reply src/pages/settings/Payments/PaymentMethodList.js:212]: I left a comment about this, this PR changed the allowed prop types (but I think it should not)

File: src/components/OfflineWithFeedback.js (hunk 9 above)
Comment (thienlnam) [reply src/components/OfflineWithFeedback.js:48]: bleh, bad merge - not intentional

File: src/components/MenuItem.js [hunk 12]
```diff
@@ -45,117 +46,122 @@ const defaultProps = {
...
-            if (props.disabled) {
-                return;
-            }
+const MenuItem = (props) => {
+    const titleTextStyle = StyleUtils.combineStyles([
+        styles.popoverMenuText,
+        styles.ml3,
+        (props.interactive && props.disabled ? styles.disabledText : undefined),
+    ], props.style);
+    const descriptionTextStyle = StyleUtils.combineStyles([styles.textLabelSupporting, styles.ml3, styles.mt1, styles.breakAll], props.style);
```
[the remark this answers is not in the document]
Comment (thienlnam) [reply src/components/MenuItem.js:55]: Ah probably not, but we can clean up the styles in a different PR

File: src/libs/PaymentUtils.js (hunk 10 above)
Comment (thienlnam) [reply src/libs/PaymentUtils.js:82]: yes, I think there was an issue to move this massaging to the PHP layer

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 11 above)
Comment (thienlnam) [reply src/pages/settings/Payments/PaymentMethodList.js:212]: I needed to update this to use `StyleUtils.combineStyles` - since we allow errorRowStyles to be either an object or an array, it started throwing an error since we were trying to use ...errorRowStyles when it was just an object

**Comment by thienlnam** [comment]
Ready for re-review

**Review by MariaHCD — DISMISSED** [review]

File: src/components/menuItemPropTypes.js [hunk 13]
```diff
@@ -10,6 +10,9 @@ const propTypes = {
     // eslint-disable-next-line react/forbid-prop-types
     wrapperStyle: stylePropTypes,
 
+    /** Used to apply offline styles to child text components */
+    style: stylePropTypes,
```
Comment (iwiznia) [inline src/components/menuItemPropTypes.js:14]: Oh! Did not know we had this, can you please use it in OfflineWithFeedback.js in lines 42 and 48?

File: src/libs/PaymentUtils.js (hunk 10 above)
Comment (iwiznia) [reply src/libs/PaymentUtils.js:82]: Have a link?

**Review by iwiznia — CHANGES_REQUESTED** [review]

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 11 above)
Comment (iwiznia) [inline src/pages/settings/Payments/PaymentMethodList.js:212]: NAB: this is more readable if you put each prop in a new line

**Review by iwiznia — DISMISSED** [review]

File: src/libs/PaymentUtils.js (hunk 10 above)
Comment (thienlnam) [reply src/libs/PaymentUtils.js:82]: Ah damn, I remembered wrong it was for policy data https://expensify.slack.com/archives/C02HWMSMZEC/p1656519994560219


I can make an issue to have PHP do the data massaging and for bank accounts as well

Comment (thienlnam) [reply src/libs/PaymentUtils.js:82]: Issue here: https://github.com/Expensify/Expensify/issues/222615

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
The X is not working for cards for me. I got the card id from the request:

image

Added the error like this:

Screen Shot 2022-08-12 at 6 14 04 PM


It looks a bit weird the card error twice. Clicking the X doesn't work:

https://user-images.githubusercontent.com/87341702/184463043-8b7e3955-d4b9-4bf3-ab13-af360d8dd004.mov

For banks, it works fine.

**Comment by thienlnam** [comment]
> It looks a bit weird the card error twice. Clicking the X doesn't work:

**Comment by aldo-expensify** [comment]
> > It looks a bit weird the card error twice. Clicking the X doesn't work:
> 
> This seems to be a bug from the cli tools utils where the card gets added to the bank list and the card list like Maria mentioned here [#10333 (review)](https://github.com/Expensify/App/pull/10333#pullrequestreview-1069500939)
> 
> I'll check it though and see if it is also happening to other cards

Ahh, I see, I'll test it with a fresh account without adding stuff with clitools

**Comment by thienlnam** [comment]
Nevermind, found the issue - it was looking for cardID instead of fundID which is why it wouldn't delete

**Comment by aldo-expensify** [comment]
You have some conflicts now, can you resolve before I do a final test 🙏

**Comment by thienlnam** [comment]
Should be good now! @aldo-expensify

File: src/libs/PaymentUtils.js (hunk 10 above)
Comment (aldo-expensify) [reply src/libs/PaymentUtils.js:82]: I may be wrong about this, but I understand that the data massaging that we are not allowing is:

Response returns data => massage data => put the data in Onyx

This is not the case here, I think we are reading this from Onyx, then process (massage) it to use it/render. This in IMO is reasonable, as I think expecting to store the data in Onyx with the exact data and structure we will be using for rendering in different places is not necessarily a good idea (we may need different things in different cases for the same data). I.e. I think here we are adding some icon/iconSize prop, this wouldn't make sense for me to come from the backend and getting added in Onyx.

**Review by aldo-expensify — APPROVED** [review]
Seems to be working well, tested in web and Android (real device)

To have the errors in Android, I found it easier to send them from the Web API adding these lines

```php
$bankAccountList = array_map(function (array $bankAccount) {
    return array_merge($bankAccount, ['pendingAction' => 'delete', 'errors' => ['1' => 'This is an error']]);
}, $bankAccountList);
$fundList = array_map(function (array $fund) {
    return array_merge($fund, ['pendingAction' => 'delete', 'errors' => ['1' => 'This is an error']]);
}, $fundList);
```

before construction `$onyxData` here: https://github.com/Expensify/Web-Expensify/blob/3b9e65f83ed4abe74e8a4bf36a84fe9cd3356f23/lib/BankAccountAPI.php#L962

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 3 above)
Comment (aldo-expensify) [reply src/pages/settings/Payments/PaymentMethodList.js]: Sounds good to me!

**Comment by thienlnam** [comment]
Going to self merge since I've addressed all reviewer comments and don't want to deal with any more merge conflicts - feel free to leave another review and I'll address in another PR

**Comment by eVoloshchak** [comment]
👋
Leaving a note that this PR didn't optimistically filter out payment methods that were pending deletion, which caused https://github.com/Expensify/App/issues/14262

**Comment by mananjadhav** [comment]
I am tagging this PR to highlight an issue fixed [here](https://github.com/Expensify/App/issues/16683). All conditions in ternary expressions or left-hand operands on conditional renders, should be boolean. This PR is one of the PRs that uses conditional render with string operands, hence I am tagging it here for the contributors to check. 

We've also updated the item in the checklist with this [PR](https://github.com/Expensify/App/pull/18069) to avoid this issues in the future.

File: src/components/MenuItem.js [hunk 14]
```diff
@@ -45,117 +46,122 @@ const defaultProps = {
...
-                    )}
-                    {props.shouldShowSelectedState && <SelectCircle isChecked={props.isSelected} />}
-                </View>
-            </>
-        )}
-    </Pressable>
-);
+                    </View>
+                    <View style={[styles.flexRow, styles.menuItemTextContainer, styles.pointerEventsNone]}>
+                        {props.badgeText && <Badge text={props.badgeText} badgeStyles={[styles.alignSelfCenter, (props.brickRoadIndicator ? styles.mr2 : undefined)]} />}
```
Comment (mananjadhav) [inline src/components/MenuItem.js:129]: @MariaHCD here is one example. But now when I searched with `badgeText` it looks like it existed before the PR too. I'll find the original PR and comment there too.
